paasta_tools/cli/cmds/validate.py

```diff
@@ -658,7 +660,8 @@ def validate_min_max_instances(service_path):
                 load_deployments=False,
                 soa_dir=soa_dir,
             )
-            if instance_config.get_instance_type() != "tron":
+            instance_config = cast(LongRunningServiceConfig, instance_config)
```

Comment:
is this required because typing the function signature made mypy type check this function?

Reply:
I have updated the return type of `get_instance_config()` to `LongRunningServiceConfig`. The only concern here is that the LONG_RUNNING_INSTANCE_TYPE_HANDLERS does not implement `adhoc` and `tron` types. I don't see any references to `load_tron_instance_config`. Not sure how to handle the `adhoc` type and if that is still necessary?

Reply:
ack

Reply:
tron and adhoc aren't in LONG_RUNNING_INSTANCE_TYPE_HANDLERS because they aren't long-running services. That is, they aren't meant to run indefinitely; tron jobs and adhoc jobs run until they complete. A long-running service like MarathonJobConfig/KubernetesServiceConfig is meant to run forever, and should get restarted when it exits. Long-running services are also the only ones that support running multiple copies, currently, which is why `get_min_instances()`/`get_max_instances()`/`get_instances()` are only defined on `LongRunningServiceConfig`, not on InstanceConfig.

I'd put the `cast()` inside the if statement, so it only gets cast if the instance type is `kubernetes`.